internal/services/cognitive/cognitive_deployment_resource.go

```diff
@@ -0,0 +1,319 @@
+	"github.com/hashicorp/terraform-provider-azurerm/internal/sdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/validation"
+)
+
+type cognitiveDeploymentModel struct {
+	Name               string                         `tfschema:"name"`
+	CognitiveAccountId string                         `tfschema:"cognitive_account_id"`
+	Model              []DeploymentModelModel         `tfschema:"model"`
+	RaiPolicyName      string                         `tfschema:"rai_policy_name"`
+	ScaleSettings      []DeploymentScaleSettingsModel `tfschema:"scale_settings"`
+}
+
+type DeploymentModelModel struct {
+	Format  string `tfschema:"format"`
+	Name    string `tfschema:"name"`
+	Version string `tfschema:"version"`
+}
+
+type DeploymentScaleSettingsModel struct {
+	ScaleType deployments.DeploymentScaleType `tfschema:"scale_type"`
+}
+
+type CognitiveDeploymentResource struct{}
+
+var _ sdk.Resource = CognitiveDeploymentResource{}
+
+func (r CognitiveDeploymentResource) ResourceType() string {
+	return "azurerm_cognitive_deployment"
+}
+
+func (r CognitiveDeploymentResource) ModelObject() interface{} {
+	return &cognitiveDeploymentModel{}
+}
+
+func (r CognitiveDeploymentResource) IDValidationFunc() pluginsdk.SchemaValidateFunc {
+	return deployments.ValidateDeploymentID
+}
+
+func (r CognitiveDeploymentResource) Arguments() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"name": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
+		},
+
+		"cognitive_account_id": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: cognitiveservicesaccounts.ValidateAccountID,
+		},
+
+		"model": {
+			Type:     pluginsdk.TypeList,
+			Required: true,
+			ForceNew: true,
+			MaxItems: 1,
+			Elem: &pluginsdk.Resource{
+				Schema: map[string]*pluginsdk.Schema{
+					"format": {
+						Type:     pluginsdk.TypeString,
+						Required: true,
+						ForceNew: true,
+						ValidateFunc: validation.StringInSlice([]string{
+							"OpenAI",
+						}, false),
+					},
+
+					"name": {
+						Type:         pluginsdk.TypeString,
+						Required:     true,
+						ForceNew:     true,
+						ValidateFunc: validation.StringIsNotEmpty,
+					},
+
+					"version": {
+						Type:     pluginsdk.TypeString,
+						Required: true,
+					},
+				},
+			},
+		},
+
+		"scale_settings": {
+			Type:     pluginsdk.TypeList,
+			Required: true,
+			ForceNew: true,
+			MaxItems: 1,
+			Elem: &pluginsdk.Resource{
+				Schema: map[string]*pluginsdk.Schema{
+					"scale_type": {
+						Type:     pluginsdk.TypeString,
+						Required: true,
+						ForceNew: true,
+						ValidateFunc: validation.StringInSlice([]string{
+							string(deployments.DeploymentScaleTypeStandard),
+						}, false),
+					},
+				},
+			},
+		},
```

Comment:
settings is redundant in tf? and scale is in the block name?
```suggestion
		"scale": {
			Type:     pluginsdk.TypeList,
			Required: true,
			ForceNew: true,
			MaxItems: 1,
			Elem: &pluginsdk.Resource{
				Schema: map[string]*pluginsdk.Schema{
					"type": {
						Type:     pluginsdk.TypeString,
						Required: true,
						ForceNew: true,
						ValidateFunc: validation.StringInSlice([]string{
							string(deployments.DeploymentScaleTypeStandard),
						}, false),
					},
				},
			},
		},
```

Reply:
@katbyte thanks for your feedback. The suggestion has been fixed. Could you please take another look?

Reply:
@katbyte thanks for your feedback. The suggestion has been fixed. Could you please take another look?